src/worldcereal/openeo/feature_extractor.py

```diff
@@ -0,0 +1,115 @@
+        "S2-L2A-B12": "B12",
+        "S1-SIGMA0-VH": "VH",
+        "S1-SIGMA0-VV": "VV",
+        "COP-DEM": "DEM",
+        "A5-tmean": "temperature-mean",
+        "A5-precip": "precipitation-flux",
+    }
+
+    def __init__(self):
+        """
+        Initializes the PrestoFeatureExtractor object, starting a logger.
+        """
+        import logging
+
+        logging.basicConfig(level=logging.INFO)
+        self.logger = logging.getLogger(PrestoFeatureExtractor.__name__)
+
+    @classmethod
+    @functools.lru_cache(maxsize=6)
+    def extract_dependencies(cls, base_url: str, dependency_name: str):
+        """Extract the dependencies from the given URL. Unpacking a zip
+        file in the current working directory.
+        """
+        import shutil
+        import urllib.request
+        from pathlib import Path
+
+        # Generate absolute path for the dependencies folder
+        dependencies_dir = Path.cwd() / "dependencies"
+
+        # Create the directory if it doesn't exist
+        dependencies_dir.mkdir(exist_ok=True, parents=True)
+
+        # Download and extract the model file
+        modelfile_url = f"{base_url}/{dependency_name}"
+        modelfile, _ = urllib.request.urlretrieve(
+            modelfile_url, filename=dependencies_dir / Path(modelfile_url).name
+        )
+        shutil.unpack_archive(modelfile, extract_dir=dependencies_dir)
+
+        # Add the model directory to system path if it's not already there
+        abs_path = str(
+            dependencies_dir / Path(modelfile_url).name.split(".zip")[0]
+        )  # NOQA
+
+        return abs_path
+
+    def output_labels(self) -> list:
+        """Returns the output labels from this UDF, which is the output labels
+        of the presto embeddings"""
+        return [f"presto_ft_{i}" for i in range(128)]
+
+    def execute(self, inarr: xr.DataArray) -> xr.DataArray:
+        import sys
+
+        if self.epsg is None:
+            raise ValueError(
+                "EPSG code is required for Presto feature extraction, but was "
+                "not correctly initialized."
+            )
+
+        # The below is required to avoid flipping of the result
+        # when running on OpenEO backend!
+        inarr = inarr.transpose("bands", "t", "x", "y")
+
+        # Change the band names
+        new_band_names = [
+            self.GFMAP_BAND_MAPPING.get(b.item(), b.item()) for b in inarr.bands
+        ]
+        inarr = inarr.assign_coords(bands=new_band_names)
+
+        # Handle NaN values in Presto compatible way
+        inarr = inarr.fillna(65535)
+
+        # Unzip de dependencies on the backend
+        self.logger.info("Unzipping dependencies")
+        deps_dir = self.extract_dependencies(self.BASE_URL, self.DEPENDENCY_NAME)
+
+        self.logger.info("Appending dependencies")
+        sys.path.append(str(deps_dir))
+
+        from dependencies.wc_presto_onnx_dependencies.mvp_wc_presto.world_cereal_inference import get_presto_features
```

Comment:
To be discussed with @HansVRP I would restructure the dependencies archive and avoid the `mvp_wc_presto` part which should not be part of the official dependencies.